war/pom.xml

```diff
@@ -367,6 +367,18 @@ THE SOFTWARE.
                   <version>1.0.4</version>
                   <type>hpi</type>
                 </artifactItem>
+                <artifactItem>
+                  <groupId>io.jenkins.plugins</groupId>
+                  <artifactId>javax-activation-api</artifactId>
+                  <version>1.2.0-1</version>
+                  <type>hpi</type>
+                </artifactItem>
+                <artifactItem>
+                  <groupId>io.jenkins.plugins</groupId>
+                  <artifactId>javax-mail-api</artifactId>
+                  <version>1.6.2-1</version>
```

Comment:
We used to bundle 1.6.5 and now we are detaching 1.6.2? https://github.com/jenkinsci/javax-mail-api-plugin/blob/javax-mail-api-1.6.2-1/pom.xml#L47 vs. https://github.com/jenkinsci/jenkins/blob/c31d9302d946463884406f920c8d14b2cf7cdf24/bom/pom.xml#L325

BTW you may consider https://www.jenkins.io/doc/developer/publishing/releasing-cd/#pom-file-modifications having plugin version automatically track library version. Otherwise it is all too easy for Dependabot to let https://github.com/jenkinsci/javax-mail-api-plugin/blob/e8b4d80f40e47e76aad413366953d333b0975f97/pom.xml#L36-L47 get out of synch.

Reply:
> We used to bundle 1.6.5 and now we are detaching 1.6.2?

Correct. Really the root cause of this problem is that the Jakarta organization published a few releases under the original package names before switching to the Jakarta package names. I suppose we _could_ release a 1.6.5 version of `javax-mail-api` that switches to the last Jakarta organization release prior to the Jakarta package rename. But then plugins wouldn't be able to depend on `javax-mail-api` and `jakarta-mail-api` concurrently without awkward Enforcer conflicts. Seemed simpler to avoid the issue by sticking to the last pre-Jakarta-organization release. I have not been able to find any meaningful differences between the last "pre-Jakarta-organization" release and the last "post-Jakarta-organization-change-pre-package-rename" release that would affect end users, much less any binary incompatibilities.

Reply:
It's a bit tough to explain, but 1.6.2 was the last release under the original `groupId` and `artifactId` (and used the `javax.` package namespace). 1.6.3, 1.6.4, and 1.6.5 were released under a new Jakarta `groupId` and `artifactId` (but still used the `javax.` package namespace). 2.0.0 was released under the same Jakarta `groupId` and `artifactId` (but used the `jakarta.` package namespace). So this makes things a bit awkward if you want to put both the old `javax.` classes and the new `jakarta.` packages on the same classpath. You can't easily do that with 1.6.3, 1.6.4, and 1.6.5 because they share the same `groupId` and `artifactId` as 2.0.0 (despite having different packages).

This doesn't matter much for production, but it does matter for plugin/BOM tests that might pull in e.g. both `javax-mail-api` and `jakarta-mail-api` as test-scoped dependencies. If I had used 1.6.5 in `javax-mail-api` that would have forced me to use the Jakarta `groupId` and `artifactId`, which would then conflict with the 2.0 version from `jakarta-mail-api`. If you want to have both sets of packages on the classpath at the same time, you basically need to stick to the original pre-Jakarta group ID and artifact ID, which also means being behind a few very small minor releases.

Just how small? See https://eclipse-ee4j.github.io/mail/docs/CHANGES.txt

Reply:
OK, makes sense now, thanks for explanation!